packages/trader/src/AppV2/Components/CurrentSpot/current-spot-display.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import clsx from 'clsx';
+import { Heading, Text } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+import { usePrevious } from '@deriv/components';
+
+type TCurrentSpotDisplayProps = {
+    has_tick_count?: boolean;
+    spot: string | null;
+    tick: number | null;
+};
+
+const ACTIONS = {
+    INC: 'increment',
+    DEC: 'decrement',
+    ADD10: 'add10',
+} as const;
+
+const CurrentSpotDisplay = ({ has_tick_count, spot, tick }: TCurrentSpotDisplayProps) => {
+    const last_digit = Number(spot?.slice(-1));
+    const prev_spot = usePrevious(spot);
+    const prev_last_digit = Number(prev_spot?.slice(-1));
+
+    const [should_enter_from_top, setShouldEnterFromTop] = React.useState(false);
+    const [should_enter_from_bottom, setShouldEnterFromBottom] = React.useState(false);
+    const [next_displayed_last_digit, setNextDisplayedLastDigit] = React.useState(last_digit);
+    const [displayed_last_digit, setDisplayedLastDigit] = React.useState(last_digit);
+
+    const last_digit_ref = React.useRef(prev_last_digit);
+    const spin_interval_id = React.useRef<ReturnType<typeof setInterval>>();
+    const spin_timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    const spinning_wrapper_ref = React.useRef<HTMLDivElement>(null);
+
+    const spinLastDigit = (action: string, interval_ms: number, start: number, end: number) => {
```

Comment:
i changed it to `string` on purpose because i was getting TS errors in `getAction()` below  but I found a way to use `keyof typeof ACTIONS`, will change it ) thanks!